### PR TITLE
Add merge state type definitions

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds merge beacon state type definitions for pkg `v3`. This follows the same pattern as `v2` and v1`

### CODE REVIEW

File: beacon-chain/state/v3/types.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// ErrNilInnerState returns when the inner state is nil and no copy set or get
+// operations can be performed on state.
+var ErrNilInnerState = errors.New("nil inner state")
+
+// BeaconState defines a struct containing utilities for the eth2 chain state, defining
+// getters and setters for its respective values and helpful functions such as HashTreeRoot().
+type BeaconState struct {
+	state                 *ethpb.BeaconStateMerge
+	lock                  sync.RWMutex
+	dirtyFields           map[types.FieldIndex]interface{}
```
Comment (potuz) [inline beacon-chain/state/v3/types.go:69]: This is `map[types.FieldIndex]bool` in v2, perhaps is better to uniformize them?

Comment (nisdas) [reply beacon-chain/state/v3/types.go:69]: +1 , also `map[types.FieldIndex]bool` avoids the inefficiencies that come with using an interface type.

**Comment by potuz** [comment]
Looks good to me. I don't know though if the unused fields will make the linter fail in every other run after this PR is merged, so will wait for nishant

**Review by potuz — APPROVED** [review]
LGTM

**Pull request merged**